L1Trigger/L1THGCal/plugins/concentrator/HGCalConcentratorProcessorSelection.cc

```diff
@@ -66,10 +76,20 @@ void HGCalConcentratorProcessorSelection::run(const edm::Handle<l1t::HGCalTrigge
       thickness = HGCalTriggerTools::kScintillatorPseudoThicknessIndex_;
     }
 
-    if (coarsenTriggerCells_ || (fixedDataSizePerHGCROC_ && thickness > kHighDensityThickness_)) {
-      coarsenerImpl_->coarsen(module_trigcell.second, trigCellVecCoarsened);
-
-      switch (selectionType_) {
+    
+    for (int subdet = 0; subdet < nSubDetectors_; subdet++){
+      
+      if ( subdet == hgcal_silicon_CEE_ && ( thickness == HGCalTriggerTools::kScintillatorPseudoThicknessIndex_ || !triggerTools_.isEm(module_trigcell.first) ) )
+        continue;
+      else if  ( subdet == hgcal_silicon_CEH_ && ( thickness == HGCalTriggerTools::kScintillatorPseudoThicknessIndex_ || triggerTools_.isEm(module_trigcell.first)  ) )
+        continue;
+      else if  ( subdet == hgcal_scintillator_ && thickness != HGCalTriggerTools::kScintillatorPseudoThicknessIndex_ )
+        continue;
```

Comment:
A loop on subdet should not be needed, the subdet can be retrieved from the module id.

Reply:
removed the loop - maybe could be simplified further?